simple/lint.go

```diff
@@ -1639,3 +1640,47 @@ func (c *Checker) LintRedundantBreak(j *lint.Job) {
 		ast.Inspect(f, fn)
 	}
 }
+
+func (c *Checker) LintStringCopy(j *lint.Job) {
+	addEmptyStr := func(x, y ast.Expr) bool {
+		bl, ok := x.(*ast.BasicLit)
+		if !ok || bl.Kind != token.STRING || bl.Value != `""` {
+			return false
+		}
+		bt, ok := j.Program.Info.TypeOf(y).(*types.Basic)
+		return ok && bt.Kind() == types.String
+	}
+	fn := func(node ast.Node) bool {
+		switch x := node.(type) {
+		case *ast.BinaryExpr: // "" + s, s + ""
+			if addEmptyStr(x.X, x.Y) || addEmptyStr(x.Y, x.X) {
+				j.Errorf(node, "redundant string copy")
```

Comment:
I would avoid all references to the idea of "redundant string copies". Copying the string may or may not be what the user had in mind. I would only point out the facts. something like the following:

`should use x instead of x + ""` and `should use x instead of string([]byte(x))` – use `j.Render` for the specific snippets of code.